app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -52,46 +53,47 @@ def destroy
   end
 
   def deploy_all
-    @stages = Project.all.flat_map do |project|
-      environment = Environment.find(params[:environment_id]) if params[:environment_id]
+    environment = deploy_group.environment
+    deploys = deploy_group.stages.map do |stage|
```

Comment:
this makes it dangerous if one of the stages does to multiple deploy groups ... should be fine for our usecase ...